common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -145,11 +144,11 @@ public abstract boolean getDeviceMode(final CommandParameters parameters)
             throws Exception;
 
     /**
-     * This method is used to determine if the QR + PIN auth flow is available on the device.
+     * This method is used to read the preferred authentication method from the broker.
      *
-     * @return true if if QR + PIN authorization is available. False otherwise.
+     * @return The preferred authentication method.
      */
-    public abstract boolean isQrPinAvailable() throws Exception;
+    public abstract PreferredAuthMethod getPreferredAuthMethod() throws Exception;
```

Comment:
Do we want to use "preferredAuthMethod" for here or using a different "preferredAuthConfiguration" to separate these two concepts (configuration from the device info and auth method sent from the parameters)

Reply:
@tongyuze 
I think those as one, if we want to separate them in to 2 concepts that's mean to have 2 types of objects to maintain with the same information.


thepreferred auth method is the desire auth method to use by the user.
and we can operate with this to:
1 we can specify this  "preferred auth method" when we make a token request.

2 we can obtained the "preferred auth method" configured by the mdm.

I think I can rename the 2 method to be more explicit.

thoughts?

Reply:
I tend to agree. Let's sync with Mihai to see if he has any concern since I saw his change separated them. https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1929/files#diff-74d19a29e3fdb6d3f69d35ea7972e9c6e2f5f45bd255e38b66213063579e6ba3

Reply:
@mipetriu 
Could these two concepts differ in the future (have different values)? The configuration is also used to setting the preferred auth method, I don't see there are big differences between them. But I do understand what you thought. I'm Ok with either option.

Reply:
Final note. We agree to have just one object.